sdk/storage/azure-storage-blob/src/samples/java/com/azure/storage/blob/ConnectionLevelTimeoutExample.java

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob;
+
+import com.azure.core.util.HttpClientOptions;
+import com.azure.storage.common.StorageSharedKeyCredential;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Locale;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * This example shows how to use read/write connection level timeouts for storage client using the Azure Storage Blob
+ * SDK for Java.
```

Comment:
"This example shows how to use connection level timeouts. These timeouts relate to discrete sections of the connection/request/response process such as time to receive headers or time between reading bytes. Please see {@link HttpClientOptions} for more detailed information on each option."